driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -431,7 +439,7 @@ CloseFuture removePool(Host host) {
         }
 
         // Wait pool creation before removing, so we don't lose connectivity
-        ListenableFuture<?> allPoolsCreatedFuture = Futures.successfulAsList(poolCreatedFutures);
+        ListenableFuture<?> allPoolsCreatedFuture = Futures.allAsList(poolCreatedFutures);
```

Comment:
There is nothing to worry about it. Only `java.lang.Error` instances can be propagated up here, all the rest is unchanged.